src/components/input/input.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  forwardRef,
+  Component,
+  HostBinding,
+  Input,
+  Output,
+  Provider,
+} from 'angular2/core';
+import {EventEmitter} from 'angular2/src/facade/async';
+import {CONST_EXPR, noop} from 'angular2/src/facade/lang';
+import {MdAriaTarget, MdAria} from '../../core/utilities/aria-directives';
+import {
+  NG_VALUE_ACCESSOR,
+  ControlValueAccessor
+} from 'angular2/src/common/forms/directives/control_value_accessor';
+import {BooleanFieldValue} from '../../core/annotations/field-value';
+import {OneOf} from '../../core/annotations/one-of';
+
+
+const MD_INPUT_CONTROL_VALUE_ACCESSOR = CONST_EXPR(new Provider(
+    NG_VALUE_ACCESSOR, {
+      useExisting: forwardRef(() => MdInput),
+      multi: true
+    }));
+
+
+let uniqueId = 0;
+
+
+/**
+ * <md-input> component.
+ */
+@Component({
+  selector: 'md-input',
+  templateUrl: 'components/input/input.html',
+  styleUrls: ['components/input/input.css'],
+  providers: [MdAria, MD_INPUT_CONTROL_VALUE_ACCESSOR],
+  directives: [MdAriaTarget],
+})
+export class MdInput implements ControlValueAccessor {
+  private _empty: boolean = true;
+  private _focused: boolean = false;
+  private _onTouched: () => void = noop;
+  private _onChangeFn: (_: any) => void = noop;
+  private _value: any = '';
+  private _inputId: number = uniqueId++;
+
+  get focused() { return this._focused; }
+  get empty() { return this._empty; }
+
+  @Input() @BooleanFieldValue() characterCounter: boolean = false;
+  @Input() @BooleanFieldValue() disabled: boolean = false;
+  @Input() @BooleanFieldValue() required: boolean = false;
+  @Input() @BooleanFieldValue() floatingPlaceholder: boolean = true;
+
+  @Input() @OneOf(['primary', 'accent']) dividerColor: string = 'primary';
+  @Input() @OneOf(['start', 'end']) align: string = 'start';
+  @Input() hintLabel: string = '';
+  @Input() maxLength: number = -1;
+  @Input() placeholder: string;
+  @Input() type: string = 'text';
+
+  get value(): any { return this._value; };
+  @Input() set value(v: any) {
+    if (this._value !== v) {
+      this._value = v;
+      this._empty = v === undefined || v === null || (''+v == '');
+      this._onChangeFn(v);
+    }
+  }
+
+  @Output('change') change = new EventEmitter(false);
+
+  @HostBinding('attr.align') private get _align(): any { return null; }
```

Comment:
Personally I'm not a fan of singe-lining getters, which I find makes the code more difficult to scan. Especially in this case where the single line has the decorator, its parameter, access modifier, name, type, and the return value.

Also this always returns null- TODO?